R/kobo-client.R

```diff
@@ -54,12 +58,45 @@ KoboClient <- R6::R6Class("KoboClient",
     #'  [`crul::HttpClient`] for reference
     #' @return Returns a list, parsed from the HttpResponse JSON object.
     get = function(path, query = list(), ...) {
-      res <- super$get(path = path, query = query, ...)
+        res <- super$get(path = path, query = query, ...)
+        # Perform additional checks, json to list parsing
+        res$raise_for_status()
+        res$raise_for_ct_json()
+        res$parse("UTF-8") %>%
+            jsonlite::fromJSON()
+    },
+
+    #' @description
+    #' Perform a POST request (with additional checks)
+    #'
+    #' @details
+    #' Extension of the `crul::HttpClient$post()` method that checks
+    #' the HttpResponse object on status, that it is of type
+    #' `application/json`, checks whether the body is a json-like string,
+    #' and if not, converts it to the json-like string, and parses the
+    #' response text subsequently from JSON to R list representation.
+    #' @param path character. Path component of the endpoint.
+    #' @param body R list or json-like string. A data payload to be sent
+    #' to the server.
+    #' @param ... crul-options. Additional option arguments, see
+    #'  [`crul::HttpClient`] for reference
+    #' @return Returns a list, parsed from the HttpResponse JSON object.
```

Comment:
Update docs, as currently the raw `response` is returned, and not a parsed list :notebook:

Reply:
done

Reply:
Thank you )